safenode/src/domain/dbc_genesis.rs

```diff
@@ -28,21 +34,28 @@ pub(super) type GenesisResult<T> = Result<T, GenesisError>;
 
 /// Main error type for the crate.
 #[derive(Error, Debug, Clone)]
-// #[non_exhaustive]
-#[non_exhaustive]
-pub(super) enum GenesisError {
-    /// Error occurred when minting the Genesis DBC.
+pub enum GenesisError {
+    /// Error occurred when creating the Genesis DBC.
     #[error("Genesis DBC error:: {0}")]
     GenesisDbcError(String),
-    /// FailedToParseReason
+    /// The dbc error reason that parsing failed.
     #[error("Failed to parse reason: {0}")]
     FailedToParseReason(#[from] DbcError),
 }
 
-/// Generate the genesis DBC.
+/// Create the genesis DBC.
 /// The genesis DBC is the first DBC in the network. It is created without
 /// a source transaction, as there was nothing before it.
-#[allow(clippy::result_large_err, unused)]
+#[allow(clippy::result_large_err)]
+pub fn create_genesis() -> GenesisResult<Dbc> {
+    let secret_key = bls::SecretKey::from_hex(GENESIS_DBC_SK).map_err(sn_dbc::Error::Blsttc)?;
+    let main_key = MainKey::new(secret_key);
+    create_genesis_dbc(&main_key)
+}
+
+/// Create a any first DBC given the key.
```

Comment:
what does `a any first DBC given the key` mean?
does it mean `create any first DBC using the given key` ?

Reply:
Yes, that's right. Will fix.

The naming difference is because it allows passing in any key, not just the hardcoded genesis key specifically.
The comment is not 100% clear..